File: backend/src/main/java/com/culcom/entity/Notice.java
package com.culcom.entity;

import com.culcom.entity.enums.NoticeCategory;
import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
@Table(name = "notices")
@Getter @Setter
@NoArgsConstructor @AllArgsConstructor
@Builder
public class Notice {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long seq;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "branch_seq", nullable = false)
    private Branch branch;

    @Column(nullable = false, length = 200)
    private String title;

    @Column(nullable = false, columnDefinition = "text")
    private String content;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    @Builder.Default
    private NoticeCategory category = NoticeCategory.공지사항;

    @Column(name = "is_pinned", nullable = false)
    @Builder.Default
    private Boolean isPinned = false;

    @Column(name = "is_active", nullable = false)
    @Builder.Default
    private Boolean isActive = true;

    @Column(name = "view_count", columnDefinition = "int unsigned default 0")
    @Builder.Default
    private Integer viewCount = 0;

    @Column(name = "event_start_date")
    private LocalDate eventStartDate;

    @Column(name = "event_end_date")
    private LocalDate eventEndDate;

    @Column(name = "created_by", length = 100)
    private String createdBy;

    @Column(name = "createdDate", nullable = false, updatable = false)
    private LocalDateTime createdDate;

    @Column(name = "lastUpdateDate")
    private LocalDateTime lastUpdateDate;

    @PrePersist
    protected void onCreate() {
        createdDate = LocalDateTime.now();
    }
}

File: frontend/src/lib/api/notice.ts
import { api, type PageResponse } from './client';
import { API } from '@/lib/routes';

export interface NoticeListItem {
  seq: number; branchName: string; title: string; category: string;
  isPinned: boolean; viewCount: number; createdBy: string; createdDate: string;
  eventStartDate: string | null; eventEndDate: string | null;
}

export interface NoticeDetail {
  seq: number; branchName: string; title: string; content: string; category: string;
  isPinned: boolean; isActive: boolean; viewCount: number; createdBy: string;
  createdDate: string; lastUpdateDate: string | null;
  eventStartDate: string | null; eventEndDate: string | null;
}

export interface NoticeCreateRequest {
  title: string; content: string; category: string; isPinned: boolean;
  createdBy?: string; eventStartDate?: string; eventEndDate?: string;
}

export interface NoticeUpdateRequest {
  title: string; content: string; category: string; isPinned: boolean;
  eventStartDate?: string; eventEndDate?: string;
}

export const noticeApi = {
  list: (params?: string) => api.get<PageResponse<NoticeListItem>>(`${API.NOTICES}${params ? `?${params}` : ''}`),
  get: (seq: number) => api.get<NoticeDetail>(API.NOTICE(seq)),
  create: (data: NoticeCreateRequest) => api.post<NoticeDetail>(API.NOTICES, data),
  update: (seq: number, data: NoticeUpdateRequest) => api.put<NoticeDetail>(API.NOTICE(seq), data),
  delete: (seq: number) => api.delete<void>(API.NOTICE(seq)),
};
